concordium-node/src/consensus_ffi/ffi.rs

```diff
@@ -1046,6 +1126,43 @@ impl ConsensusContainer {
         Ok((out_hash, out_data))
     }
 
+    /// Get the best guess as to what the next account sequence number should
+    /// be. If all account transactions are finalized, then this information
+    /// is reliable. Otherwise, this is the best guess, assuming all other
+    /// transactions will be committed to blocks and eventually finalized.
+    pub fn get_next_account_sequence_number_v2(
+        &self,
+        account_identifier: &crate::grpc2::types::AccountIdentifierInput,
+    ) -> Result<Vec<u8>, tonic::Status> {
+        use crate::grpc2::Require;
+        let (acc_type, acc_id) =
+            crate::grpc2::types::account_identifier_to_ffi(account_identifier).require()?;
+        let consensus = self.consensus.load(Ordering::SeqCst);
+        let mut out_data: Vec<u8> = Vec::new();
+        let response: ConsensusQueryResponse = unsafe {
+            getNextAccountSequenceNumberV2(
+                consensus,
+                acc_type,
+                acc_id,
+                &mut out_data,
+                copy_to_vec_callback,
+            )
+            .try_into()?
+        };
+        response.ensure_ok("account")?;
+        Ok(out_data)
+    }
+
+    /// Get information of the current state of consensus.
+    pub fn get_consensus_info_v2(&self) -> Result<Vec<u8>, tonic::Status> {
+        let consensus = self.consensus.load(Ordering::SeqCst);
+        let mut out_data: Vec<u8> = Vec::new();
+        let _response: ConsensusQueryResponse = unsafe {
+            getConsensusInfoV2(consensus, &mut out_data, copy_to_vec_callback).try_into()?
+        };
```

Comment:
Why not check the response? I suppose it should never fail, but please document this.

Reply:
I have added a comment